### ISSUE TITLE
Flickering on resize

### ISSUE THREAD

**Issue by ken-kuro** [warpdotdev/Warp#4821]
### Describe the bug

Resizing windows gives a lot of black flickering around the application. It happens since the latest `stable_03` release (it's easier to see in light theme because of the larger contrast ration)

### To reproduce

1. Install the latest version which is `v0.2024.04.16.08.02.stable_03`
2. (Optional) Switch to a light theme for better contrast
3. Resize the window and notice how it flickers to black

### Expected behavior

The application resizes without any flickering

### Screenshots

[Screencast from 2024-04-24 09-07-28.webm](https://github.com/warpdotdev/Warp/assets/47441476/2a7a054e-274a-453f-8f14-dcbc0259cca6)
[Screencast from 2024-04-24 09-08-09.webm](https://github.com/warpdotdev/Warp/assets/47441476/8add60cb-23bb-4140-a86a-efb388db4d78)

### Operating system

Linux

### Operating system and version

Arch Linux 6.8.7-arch1-1

### Shell Version

fish, version 3.7.1

### Current Warp version

v0.2024.04.16.08.02.stable_03

### Regression

Yes, this bug started recently or with an X Warp version

### Recent working Warp date

v0.2024.04.16.08.02.stable_02

### Additional context

I tried to reproduce it with version `stable_02` and the issue only occurs with the latest update, so I believe this is a regression.
Here're the logs:
[eglinfo.txt](https://github.com/warpdotdev/Warp/files/15103097/eglinfo.txt)
[warp-terminal-stable-02.txt](https://github.com/warpdotdev/Warp/files/15103101/warp-terminal-stable-02.txt)
[warp-terminal-stable-03.txt](https://github.com/warpdotdev/Warp/files/15103102/warp-terminal-stable-03.txt)

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @ken-kuro We've released a fix in the latest but from today which may help with the issue. Please note this related issue here: https://github.com/warpdotdev/Warp/issues/4809#issuecomment-2075450340

**Comment by ken-kuro** [comment]
Unfortunately the issue still exists on latest patch, here's the log:
[warp-terminal-v0.2024.04.23.08.01.stable_03.txt](https://github.com/warpdotdev/Warp/files/15124318/warp-terminal-v0.2024.04.23.08.01.stable_03.txt)

**Comment by MrBanja** [comment]
Got the same problem. Makes it hard to use some TUIs (superfile in my example)

![CleanShot 2024-05-10 at 23 00 16](https://github.com/warpdotdev/Warp/assets/18736922/adaf35b7-ee76-4521-a277-dd61e087d674)

**Comment by dannyneira** [comment]
Hi @MrBanja @ken-kuro on the [2024.05.02 (v0.2024.04.30.08.02) release](https://docs.warp.dev/getting-started/changelog#id-2024.05.02-v0.2024.04.30.08.02) we improved some of the flickering issues. Make sure to [download the latest version](https://warp.dev/linux-terminal)

Please let me know if any of those helped or if you still see the issue.

**Comment by MrBanja** [comment]
Hi @dannyneira! Thank you for your response. 

I am running a v0.2024.05.07.08.02.stable_02

**Comment by ken-kuro** [comment]
> Hi @MrBanja @ken-kuro on the [2024.05.02 (v0.2024.04.30.08.02) release](https://docs.warp.dev/getting-started/changelog#id-2024.05.02-v0.2024.04.30.08.02) we improved some of the flickering issues. Make sure to [download the latest version](https://warp.dev/linux-terminal)
> 
> Please let me know if any of those helped or if you still see the issue.

Sadly the problem still occurs on that latest version :<

**Comment by dannyneira** [comment]
I'll be sure to let the team know of this possible regression. We'll update this thread with any changes, thanks for your patience folks.

**Comment by ken-kuro** [comment]
Just want to update the current status on the latest version `v0.2024.06.04.08.02.stable_02`. There's no black flickering anymore, but it looks like it flicker the content behind it (the wallpaper in my case). Since I'm using arch, so there's might be something relates to the graphic driver being updated as well.
[Screencast from 2024-06-07 10-26-29.webm](https://github.com/warpdotdev/Warp/assets/47441476/28f0eb32-f970-46d6-87cd-999c32a691b9)
